tests/integration/integration_python_test.go

```diff
@@ -1161,6 +1161,27 @@ func TestPythonAwaitOutputs(t *testing.T) {
 			},
 		})
 	})
+
+	// This checks we don't leak futures awaiting outputs. Regression test for
+	// https://github.com/pulumi/pulumi/issues/16055.
+	//
+	//nolint:paralleltest // ProgramTest calls t.Parallel()
+	t.Run("OutputLeak", func(t *testing.T) {
+		// Skip if python is < 3.9
+		version, err := exec.Command("python3", "--version").Output()
+		require.NoError(t, err)
+		if strings.Contains(string(version), "3.8") {
+			t.Skip("Skipping test as Python version is < 3.9")
+		}
```

Comment:
Nit: Mention _why_ we skip on < 3.9 (use of `asyncio.to_thread()` which is 3.9+)?